airbyte-webapp/src/hooks/services/Connection/ConnectionFormService.test.tsx

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { act } from "@testing-library/react";
+import { renderHook } from "@testing-library/react-hooks";
+import React from "react";
+import { MemoryRouter } from "react-router-dom";
+
+import { WebBackendConnectionRead } from "core/request/AirbyteClient";
+import { TestWrapper } from "utils/testutils";
+
+import { ModalCancel } from "../Modal";
+import {
+  ConnectionFormServiceProvider,
+  ConnectionServiceProps,
+  useConnectionFormService,
+} from "./ConnectionFormService";
+import mockConnection from "./mockConnection.json";
+import mockDest from "./mockDestinationDefinition.json";
+
+const mockWorkspace = {
+  workspaceId: "47c74b9b-9b89-4af1-8331-4865af6c4e4d",
+  customerId: "55dd55e2-33ac-44dc-8d65-5aa7c8624f72",
+  email: "krishna@airbyte.com",
+  name: "47c74b9b-9b89-4af1-8331-4865af6c4e4d",
+  slug: "47c74b9b-9b89-4af1-8331-4865af6c4e4d",
+  initialSetupComplete: true,
+  displaySetupWizard: false,
+  anonymousDataCollection: false,
+  news: false,
+  securityUpdates: false,
+  notifications: [],
+};
+
+["packages/cloud/services/workspaces/WorkspacesService", "services/workspaces/WorkspacesService"].forEach((s) =>
+  jest.mock(s, () => ({
+    useCurrentWorkspaceId: () => mockWorkspace.workspaceId,
+    useCurrentWorkspace: () => mockWorkspace,
+  }))
+);
+
+jest.mock("../FormChangeTracker", () => ({
+  useFormChangeTrackerService: () => ({ clearFormChange: () => null }),
+  useUniqueFormId: () => "blah",
+}));
+jest.mock("services/connector/DestinationDefinitionSpecificationService", () => ({
+  useGetDestinationDefinitionSpecification: () => mockDest,
+}));
+
+describe("ConnectionFormService", () => {
```

Comment:
I'd like to see more testing here. This only tests for onSubmit, but there are 11 other fields in the service

Reply:
This is mostly just utilizing other hooks and passing those values along. I'm happy to discuss but I feel like testing those means writing tests for just those hooks, not this service.

Reply:
Are those hooks tested?  I think we want to make sure this entire workflow has good coverage.

Reply:
Added tests to most of these hooks.